File: src/p006/p006.py
#!/usr/bin/env python3


def sum_of_squares(n):
    return n * (n + 1) * (2 * n + 1) // 6


def square_of_sum(n):
    return (n * (n + 1) // 2) ** 2


def series_solution(n):
    return square_of_sum(n) - sum_of_squares(n)


if __name__ == "__main__":
    print(series_solution(100))

File: src/p006/p006.lua
#!/usr/bin/env lua

local function seriesSolution()
  local function sumOfSquares(n)
    return n * (n + 1) * (2 * n + 1) // 6
  end

  local function squareOfSum(n)
    return (n * (n + 1) // 2) ^ 2
  end

  return squareOfSum(100) - sumOfSquares(100)
end

print(seriesSolution())
